fars/api/views.py

```diff
@@ -24,6 +27,65 @@ class BookingsList(viewsets.ViewSetMixin, generics.ListAPIView):
     filter_backends = (filters.DjangoFilterBackend,)
     filter_class = BookingFilter
 
+    def filter_on_bookable(self, bookings, request):
+        bookable = request.query_params.get('bookable')
+        if bookable:
+            return bookings.filter(bookable__id_str=bookable), True
+        return bookings, False
+
+    def serialize_one(self, booking):
+        # If the booking is None, the serializer will create a null object, which we don't want
+        return Response(self.get_serializer_class()(booking).data if booking else None)
+
+    def serialize_many(self, bookings):
+        # If the booking is None, the serializer will create a null object, which we don't want
+        return Response(self.get_serializer_class()(bookings, many=True).data)
+
+    def filter_on_bookable_and_serialize_many(self, bookings, request):
+        bookings, _ = self.filter_on_bookable(bookings, request)
+        return self.serialize_many(bookings)
+
+    @action(methods=['get'], detail=False)
+    def future(self, request):
```

Comment:
It's good practice to name methods like these after a verb, e.g. `getFutureBookings`, `getPastBookings`, `getNextBooking`, `getPreviousBooking` and `getCurrentBookings`